### PR TITLE
[wip] chore: adjust compile interface

### PR DESCRIPTION
**Pull request by nom4dv3**

### CODE REVIEW

File: src/api/compile.ts [hunk 1]
```diff
@@ -214,24 +219,29 @@ export async function compileServer(
...
+    data.append('wasmFile', wasmFile)
+    const yamlFile = new File([new Blob([cleYaml.toString()], { type: 'text/yaml' })], 'src/zkgraph.yaml', { type: 'text/yaml' })
+    data.append('yamlFile', yamlFile)
+  }
+  else {
+    const tmpPath = path.join(tmpdir(), randomStr())
+    fs.writeFileSync(tmpPath, cleYaml.toString())
+    data.append('wasmFile', fs.createReadStream(outWasmPath))
+    data.append('yamlFile', fs.createReadStream(tmpPath))
+  }
```
[the remark this answers is not in the document]
Comment (nom4dv3) [reply src/api/compile.ts:236]: can we create a readstream directly from the uint8array without touching the file system?
e.g.
```ts
import { Readable } from 'stream';

// Function to create a readable stream from a Uint8Array
function createReadStreamFromArray(array: Uint8Array): Readable {
    return new Readable({
        read() {
            // Push data chunks from the Uint8Array
            this.push(array);
            this.push(null); // Signal the end of the stream
        }
    });
}
```

File: src/api/compile.ts [hunk 2]
```diff
@@ -214,24 +219,29 @@ export async function compileServer(
...
-  // @murong: TODO: try fill 'data' with wasmUint8Array, cleYaml
   // Set up form data
   const data = new FormData()
-  // data.append('wasmFile', fs.createReadStream(tmpWasmPath))
-  // data.append('yamlFile', fs.createReadStream(yamlPath))
+  if (__BROWSER__) {
+    const blob = new Blob([wasmUint8Array], { type: 'application/wasm' })
+    const wasmFile = new File([blob], 'inner.wasm', { type: 'application/wasm' })
+    data.append('wasmFile', wasmFile)
+    const yamlFile = new File([new Blob([cleYaml.toString()], { type: 'text/yaml' })], 'src/zkgraph.yaml', { type: 'text/yaml' })
```
[the remark this answers is not in the document]
Comment (nom4dv3) [reply src/api/compile.ts:228]: i've fixed this in this commit: https://github.com/hyperoracle/zkgraph-api/commit/ca0843fb743e95e2a560c6e68e1f1e6ce29db887

File: src/api/compile.ts [hunk 3]
```diff
@@ -171,3 +282,10 @@ export async function compileRequest(endpoint: string, data: any) {
 
   return await axios.request(requestConfig)
 }
+
+function createOnNonexist(filePath: string): void {
+  const directoryPath = path.dirname(filePath)
+
+  if (!fs.existsSync(directoryPath))
+    fs.mkdirSync(directoryPath, { recursive: true })
+}
```
[the remark this answers is not in the document]
Comment (murongg) [reply src/api/compile.ts:291]: automatically remove here by tree shaking

File: src/api/compile.ts [hunk 4]
```diff
@@ -1,17 +1,17 @@
-import { hasOwnProperty, randomStr } from '@murongg/utils'
+import fs from 'node:fs'
+import path from 'node:path'
+import { tmpdir } from 'node:os'
```
[the remark this answers is not in the document]
Comment (murongg) [reply src/api/compile.ts:3]: no , here can't use `!BROWSER`, automatically remove here by tree shaking

File: src/api/compile.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (murongg) [reply src/api/compile.ts:236]: look good, i will try it.

**Pull request merged**